src/Oscoin/Deployment/Node/Options.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE DataKinds #-}
+
+module Oscoin.Deployment.Node.Options where
+
+import           Oscoin.Prelude hiding (option)
+
+import           Oscoin.Configuration (ConfigPaths)
+import           Oscoin.Crypto.PubKey (PublicKey)
+import qualified Oscoin.Node.Options as Node
+import qualified Oscoin.P2P.Disco.Options as Disco
+
+import           Codec.Serialise (Serialise)
+import           Options.Applicative
+
+data CloudProvider
+    = GCE GceOptions
+    deriving Show
+
+data GceOptions = GceOptions
+    { gceNumSSDs :: Word8
+    } deriving Show
+
+cloudProviderParser :: Parser CloudProvider
+cloudProviderParser = hsubparser $
+    command "gce"
+            (info (GCE <$> gceOpts) (progDesc "Target Google Compute Engine"))
+  where
+    gceOpts = GceOptions
+        <$> option (map (max 1 . min 8) auto)
+            ( long    "num-ssds"
+           <> help    "Number of local SSDs to provision (min 1, max 8)"
+           <> metavar "INT"
+           <> value   1
+           <> showDefault
+            )
+
+data ConfigFormat
+    = CloudInit
+    deriving Show
+
+readConfigFormat :: String -> Either String ConfigFormat
+readConfigFormat = \case
+    "cloud-init" -> pure CloudInit
+    x            -> Left $ "Unsupported configuration format: " <> x
+
+data Options crypto network = Options
+    { optVersion     :: Text
+    , optProvider    :: CloudProvider
+    , optFormat      :: ConfigFormat
+    , optOutput      :: Maybe FilePath
+    , optNodeOptions :: Node.Options crypto network
+    } deriving Generic
+
+deriving instance (Show (PublicKey c), Show n) => Show (Options c n)
+
+nodeDeployOptionsParser
+    :: Serialise (PublicKey c)
+    => ConfigPaths
+    -> Parser (Options c Disco.OptNetwork)
+nodeDeployOptionsParser cps = Options
+    <$> option str
+        ( long    "version"
+       <> help    "Version to deploy. Should be commit SHA1."
+       <> metavar "STRING"
+        )
+    <*> cloudProviderParser
+    <*> option (eitherReader readConfigFormat)
+        ( long    "format"
+       <> help    "Machine configuration format"
+       <> value   CloudInit
```

Comment:
This shows `(default: CloudInit)` in the help text but the parser only accepts `cloud-init`. It would also be great if we could document the possible formats, i.e. only `cloud-init` at the moment.